test/rules/prefer_interpolation_to_compose_strings.dart

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `pub run test -N prefer_interpolation_to_compose_strings`
+
+void main() {
+  String name = 'Juan'+'Pablo'; // OK // It is not the business of this rule
+  int year = 2017;
+  int birth = 1993;
+  'Hello, $name! You are ${year - birth} years old.';
+  'Hello, ' + // LINT
+      name + // LINT
+      '! You are ' + // LINT
+      (year - birth).toString() + // LINT
+      ' years old.';
```

Comment:
Wow! That's a lot of lints for a single expression. Perhaps ignore lints in subexpressions.